### DISCUSSION TITLE
Does remix need backend api service? What I want to do is SSR mode.

### DISCUSSION THREAD

**Discussion by Alvazz** [remix-run/remix#4653] [Q&A]
I have a lot of doubts about remix. I said that I am a back-end developer in Go and a novice in front-end development. What I want to ask is, is it necessary to use remix to write back-end api for remix to call? Is that so? Or am I understanding it wrong? Please give me some guidance! thank you very much!

**Comment by kiliman** [comment]
If you want to write your backend API in Go, you can simply use Remix as a front-end-to-backend framework.

Your route loaders/actions would `fetch` your Go APIs and return the data to your route.

```ts
// routes/index.ts
export const loader = async ({request}: LoaderArgs) => {
  const payload = {} // some payload for your API
  return fetch('http://goserver/api/get-data', payload)
}

export default function Index() {
  const data = useLoaderData() // data from your Go API
}
```
